### PR TITLE
[VBLOCKS-1502] Detox pipeline

### PR DESCRIPTION
**Pull request by mhuynh5757**
Add a Detox pipeline to the Reference App

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: app/index.js
Comment (mhuynh5757) [inline app/index.js:1]: I had to do this to get the bundler detecting the entry point of the project correctly. Unfortunate, but there's not a lot of heavy lifting we do in this file anyways.

Comment (mhuynh5757) [reply app/index.js:1]: It should, and Android actually supports this. The issue arises when the bundler is run in `release` mode (again). Android has an option to define a different entry point file in the `app/build.gradle` file, but iOS doesn't seem to have an analogous option that I could find.

Comment (mhuynh5757) [reply app/index.js:1]: The bundler is specifically looking for `index.js` when in `release` mode.

**Review by kbagchiGWC — COMMENTED** [review]
Android changes looks fine.

**Review by charliesantos — APPROVED** [review]
Great work @mhuynh5757

**Review by charliesantos — APPROVED** [review]

**Pull request merged**